api/lcp/credential.py

```diff
@@ -0,0 +1,47 @@
+import uuid
+
+from core.model import Collection, Credential, DataSource
+
+
+class CredentialFactory(object):
+    """Generates patron's credentials used by the LCP License Server"""
+
+    IDENTIFIER_FROM_REMOTE_SERVICE = 'User identifier passed to LCP server'
+
+    def __init__(self, collection):
+        """Initializes a new instance of CredentialFactory class
+
+        :param collection: Collection object
+        :type collection: Collection
+        """
+        self._collection_id = collection.id
+
+    def create(self, db, patron):
+        """Generates patron's credentials
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :param patron: Patron object
+        :type patron: core.model.patron.Patron
+        """
+        collection = Collection.by_id(db, id=self._collection_id)
+
+        def _refresh_credential(credential):
+            credential.expires = None
+            credential.credential = str(uuid.uuid4())
+
+            return credential
+
+        credential = Credential.lookup(
+            db,
+            DataSource.LCP,
+            self.IDENTIFIER_FROM_REMOTE_SERVICE,
```

Comment:
This makes it look like the credential is coming _from_ the LCP license server rather than being generated here and sent _to_ the license server; see my earlier comment for my suggested strings.

Also, I think you can replace this whole class with `Credential.persistent_token_create`; do you need anything different from that?

Reply:
I added two methods: 
- one for generating patron IDs which is public information and included into LCP licenses 
- the other one for generating passphrases which is private information